### ISSUE TITLE
You can repeatedly add keys and it will interpret them as string

### ISSUE THREAD

**Issue by lingtjien** [Tencent/rapidjson#1349]
```
#include <rapidjson/document.h>
#include "rapidjson/writer.h"
#include "rapidjson/prettywriter.h"
#include "rapidjson/stringbuffer.h"

#include <iostream>

int main ()
{
  rapidjson::StringBuffer body;
  rapidjson::PrettyWriter<rapidjson::StringBuffer> json(body);
  
  json.StartObject();
  json.Key("test");
  json.Key("test2");
  json.Key("test3");
  json.Key("test4");
  json.EndObject();
  
  std::cout << body.GetString() << std::endl;

  return 0;
}
```

when executed prints:

```
{
    "test": "test2",
    "test3": "test4"
}
```

whereas I would expect the correct code to generate this result to be:
```
  json.StartObject();
  json.Key("test");
  json.String("test2");
  json.Key("test3");
  json.String("test4");
  json.EndObject();
```

**Comment by miloyip** [comment]
So do you expect an assertion failure?

**Comment by lingtjien** [comment]
yes, it seems wrong to me that you can keep adding keys while it actually adds string values. The word "Key" would then be misleading as it can currently also be used for "String"

**Comment by miloyip** [comment]
The original thought was to make it more clearly for writing objects as key-value paired calls. I think your concern is right, but adding additional check may incur overheads even for non-debug build. I need to consider about how assertion should be implemented.

**Comment by Bu11etmagnet** [comment]
After Key is called, a second call to Key should result in an error (e.g. throwing a std::logic_error).

Writer should have an internal state that controls which methods are allowed in the current state. If you called Key, next you have to call StartObject or one of the value methods (Null, String, Int, ...).

See for example https://github.com/stleary/JSON-java/blob/master/JSONWriter.java
